examples/remix/app/routes/buckets/index.tsx

```diff
@@ -0,0 +1,58 @@
+import { ActionFunction, Form, useLoaderData } from "remix";
+import { s3Client } from "../../../backend/s3Client.server";
+import { createBucket, listBuckets } from "../../../backend/buckets.server";
+
+type BucketData = {
+    name: string;
+    createdAt: string;
+}
+
+type BucketsList = {
+    buckets: BucketData[];
+}
+
+// action is executed on the server on incoming POST requests
+export const action: ActionFunction = async ({ request }) => {
+    let { name } = Object.fromEntries(await request.formData());
+
+    if (typeof name !== "string") {
+        throw new Response("Invalid data submitted", { status: 400 });
+    }
+
+    await createBucket(s3Client, { Bucket: name as string });
+    console.log("created bucket", name);
+    return "success";
+};
+
+// loader is executed on the server for GET requests
+// the result of the loader is accessed using `useLoaderData` on the client
+export async function loader(): Promise<BucketsList> {
+    const buckets = await listBuckets(s3Client, {});
+    return {
+        buckets: buckets.map(bucket => ({ name: bucket.Name ?? 'unknown', createdAt: bucket.CreationDate?.toISOString() ?? "unknown" }))
```

Comment:
The bucket's name should be lowercased looking at its type: `bucket.name`

As for the date, shouldn't it be `createdAt` and not `CreationDate`?

Regarding the string "unknown" here, wouldn't undefined be suitable here if a property doesn't exist?